src/test/java/org/alfresco/repo/event2/UpdateRepoEventIT.java

```diff
@@ -820,4 +827,62 @@ public void testCreateAndMoveFileInTheSameTransaction()
 
         assertEquals("Wrong node parent.", folder2ID, moveFileParentAfterMove);
     }
+
+    @Test
+    public void testAddAspectRemoveAspectFromContentSameTransactionTest()
+    {
+        final NodeRef nodeRef = createNode(ContentModel.TYPE_CONTENT);
+        NodeResource resource = getNodeResource(1);
+        final Set<String> originalAspects = resource.getAspectNames();
+        assertNotNull(originalAspects);
+
+
+        retryingTransactionHelper.doInTransaction(() -> {
+            // Add cm:geographic aspect with default value
+            nodeService.addAspect(nodeRef, ContentModel.ASPECT_GEOGRAPHIC, null);
```

Comment:
I think this test will fail if we change it to also include property. E.g. `nodeService.addAspect(nodeRef, ContentModel.ASPECT_GEOGRAPHIC, Map.of(ContentModel.PROP_LATITUDE, 52.53122d));`

Anyway I just left comment here to remind myself later, it is not a blocker for this PR.